examples/computer_vision/cifar10_pytorch_inference/model_def.py

```diff
@@ -0,0 +1,95 @@
+"""
+Perform inference on pretrained CIFAR10 from https://github.com/huyvnphan/PyTorch_CIFAR10
+"""
+
+import tempfile
+from typing import Any, Dict, Sequence, Tuple, Union, cast
+
+import torch
+import torchvision
+from torch import nn
+from torchvision import transforms
+import torchvision.models as models
+
+from determined.pytorch import DataLoader, PyTorchTrial, PyTorchTrialContext
+import resnet
+
+# Constants about the data set.
+IMAGE_SIZE = 32
+NUM_CHANNELS = 3
+NUM_CLASSES = 10
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+
+
+def accuracy_rate(predictions: torch.Tensor, labels: torch.Tensor) -> float:
+    """Return the accuracy rate based on dense predictions and sparse labels."""
+    assert len(predictions) == len(labels), "Predictions and labels must have the same length."
+    assert len(labels.shape) == 1, "Labels must be a column vector."
+
+    return (  # type: ignore
+        float((predictions.argmax(1) == labels.to(torch.long)).sum()) / predictions.shape[0]
+    )
+
+class CIFARTrial(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext) -> None:
+        self.context = context
+        self.download_directory = tempfile.mkdtemp()
+
+        # TODO: Load your trained model
+        self.model = self.context.wrap_model(resnet.resnet18(pretrained=True))
```

Comment:
i don't think the `resnet.py` file and the predownloaded checkpoint are necessary.  

instead, let's follow the steps here to finetune a pretrained model from torchvision so we don't need to include model code and checkpoint in this example.
https://colab.research.google.com/github/pytorch/tutorials/blob/gh-pages/_downloads/df1f5ef1c1a8e1a111e88281b27829fe/finetuning_torchvision_models_tutorial.ipynb#scrollTo=n4pbCfUO0wWs

Reply:
Yeah, I agree that it'd be nice if we didn't have to upload a checkpoint. If we did it this way, wouldn't the last layer be untrained and therefore give us bad predictions? Or maybe it's good enough for the purposes for an example.

Reply:
the last layer was trained to classify imagenet classes so we need to remove that layer and add a layer for cifar10 classes.

Reply:
Yeah, I'm just wondering if it's a problem that the new cifar10 classifier layer is untrained. I can just try it at see what the accuracy is

Reply:
@liamcli I've used the pretrained imagenet in the most recent commit. However, without additional finetuning, this gets a validation error of 88%. With the CIFAR10-trained checkpoint we got an error of ~6%. Do you think we should revert and go back to uploading a large checkpoint or go with the unfinetuned model? I could also use mobilenet pretrained on cifar10 if we wanted to upload a _smaller_ checkpoint.